src/object_store/src/object/mem.rs

```diff
@@ -254,25 +233,34 @@ impl InMemObjectStore {
         *SHARED.lock() = InMemObjectStore::new();
     }
 
-    async fn get_object<R, F>(&self, path: &str, f: F) -> ObjectResult<R>
-    where
-        F: Fn(&Bytes) -> R,
-    {
-        self.objects
-            .lock()
-            .await
+    async fn get_object(
+        &self,
+        path: &str,
+        range: impl RangeBounds<usize> + Clone + Send + Sync + 'static,
```

Comment:
src/storage/src/store.rs  `StaticSendSync`

Reply:
`range` here requires a lot more trait bounds, IMO, expend them here is also okay.